Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/PatientDiscoveryTransforms.java

```diff
@@ -0,0 +1,220 @@
+            PRPAIN201306UV02 response = (PRPAIN201306UV02) msg;
+            List<II> oII = getPatientIdFromResponse(response);
+            if (oII != null && oII.size() > 0) {
+                for (II entry : oII) {
+                    if (entry != null && entry.getRoot() != null && entry.getExtension() != null
+                        && !entry.getRoot().isEmpty() && !entry.getExtension().isEmpty()) {
+                        createPatientParticipantObjectIdentification(auditMsg, entry.getRoot(), entry.getExtension());
+                    }
+                }
+            } else {
+                createPatientParticipantObjectIdentification(auditMsg, null, null);
+            }
+        }
+        return auditMsg;
+    }
+
+    private AuditMessageType getQueryParamsParticipantObjectIdentification(Object msg, AuditMessageType auditMsg)
+        throws JAXBException {
+
+        ParticipantObjectIdentificationType participantObject = new ParticipantObjectIdentificationType();
+        participantObject = createParticipantObjectIdentification(
+            PatientDiscoveryTransformConstants.PARTICIPANT_QUERYPARAMS_OBJ_TYPE_CODE_SYSTEM,
+            PatientDiscoveryTransformConstants.PARTICIPANT_QUERYPARAMS_OBJ_TYPE_CODE_ROLE,
+            PatientDiscoveryTransformConstants.PARTICIPANT_QUERYPARAMS_OBJ_ID_TYPE_CODE,
+            PatientDiscoveryTransformConstants.PARTICIPANT_QUERYPARAMS_OBJ_ID_TYPE_CODE_SYSTEM,
+            PatientDiscoveryTransformConstants.PARTICIPANT_QUERYPARAMS_OBJ_ID_TYPE_DISPLAY_NAME);
+        participantObject.setParticipantObjectID(createUUID());
+        participantObject.setParticipantObjectName(HomeCommunityMap.
+            formatHomeCommunityId(HomeCommunityMap.getLocalHomeCommunityId()));
+        byte[] byteArray = getParticipantObjectQuery(msg);
+        if (byteArray != null) {
+            participantObject.setParticipantObjectQuery(byteArray);
+        }
+        auditMsg.getParticipantObjectIdentification().add(participantObject);
+        return auditMsg;
+
+    }
+
+    private byte[] getParticipantObjectQuery(Object msg) throws JAXBException {
+        PRPAIN201305UV02 request = null;
+        PRPAIN201305UV02 response = null;
+        JAXBContextHandler oHandler = new JAXBContextHandler();
+        String JAXB_HL7_CONTEXT_NAME = "org.hl7.v3";
+        JAXBContext jc = oHandler.getJAXBContext(JAXB_HL7_CONTEXT_NAME);
+        Marshaller marshaller = jc.createMarshaller();
+        ByteArrayOutputStream baOutStrm = new ByteArrayOutputStream();
+        baOutStrm.reset();
+        if (msg instanceof PRPAIN201305UV02) {
+            request = (PRPAIN201305UV02) msg;
+            if (request.getControlActProcess() != null && request.getControlActProcess().
+                getQueryByParameter() != null) {
+                marshaller.marshal(request.getControlActProcess().getQueryByParameter(), baOutStrm);
+            }
+        } else if (msg instanceof PRPAIN201305UV02) {
```

Comment:
This "else" is the same condition as the original "if" -- not sure what it's supposed to be, but take a look and fix this.